### PR TITLE
Align our code with go-client master branch

### LINKED ISSUE #435: Align our code with go-client master branch
Change of go-client version to master under `glide.yaml` will raise errors on the `make` stage.
```
# kubevirt.io/kubevirt/pkg/api/v1
api/v1/types.go:67: cannot use VM literal (type *VM) as type runtime.Object in argument to scheme.AddKnownTypes:
        *VM does not implement runtime.Object (missing DeepCopyObject method)
api/v1/types.go:68: cannot use VMList literal (type *VMList) as type runtime.Object in argument to scheme.AddKnownTypes:
        *VMList does not implement runtime.Object (missing DeepCopyObject method)
api/v1/types.go:71: cannot use Spice literal (type *Spice) as type runtime.Object in argument to scheme.AddKnownTypes:
        *Spice does not implement runtime.Object (missing DeepCopyObject method)
api/v1/types.go:72: cannot use Migration literal (type *Migration) as type runtime.Object in argument to scheme.AddKnownTypes:
        *Migration does not implement runtime.Object (missing DeepCopyObject method)
api/v1/types.go:73: cannot use MigrationList literal (type *MigrationList) as type runtime.Object in argument to scheme.AddKnownTypes:
        *MigrationList does not implement runtime.Object (missing DeepCopyObject method)
api/v1/types.go:85: unknown field 'ImportPrefix' in struct literal of type announced.GroupMetaFactoryArgs
api/v1/types.go:171: cannot use &vl.ListMeta (type *"kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/apis/meta/v1".ListMeta) as type "kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/apis/meta/v1".List in return argument
api/v1/types.go:470: cannot use &ml.ListMeta (type *"kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/apis/meta/v1".ListMeta) as type "kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/apis/meta/v1".List in return argument
```

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Fixes #435 
Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

### CODE REVIEW

**Comment by rmohr** [comment]
ok to test

**Comment by rmohr** [comment]
@stu-gott could you have a look? Might be interesting for your api-aggregation work.

**Comment by rmohr** [comment]
@cynepco3hahue could it be, that the provided glide.lock is not the final one? I see that on travis, which does a sync based on glide.yaml, it works, whereas on jenkins, where we only use glide.lock, it fails ...

**Comment by cynepco3hahue** [comment]
@rmohr strange does it possible that the vagrant VM has some remainings?
I can see under the jenkins log
```
==> node0: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> node0: flag to force provisioning. Provisioners marked to run always will still run.
=-=> master: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> master: flag to force provisioning. Provisioners marked to run always will still run.
```
so looks like we do not start clear VM each time, also under the tear down I can see 
`vagrant halt` I think we must destroy via `vagrant destroy`
or at least we can run `make distclean`

**Comment by rmohr** [comment]
The vagrant vms are not involved. They are just already started before we compile the code. The code is compiled on the jenkins slave itself. Normally glide should be able to substitute the vendor folder completely based on the glide.lock file. Could you just check, that after a make on latest master, and then doing a make on your branch everything works? If not, you probably have to do another "make sync", and commit the resulting glide.lock.

And yes, we could do a `make distclean` on every run. Still the current output might be a hint for something being not right in glide.lock.

**Comment by cynepco3hahue** [comment]
@rmohr Ok I checked two scenarios

first scenario
====================
- change branch to master
- run `make distclean`
- run `make`
- change to my branch
- run `make`
In result a lot of errors

second scenario
====================
- change branch to master
- run `make distclean`
- run `make`
- change to my branch
- run `make distclean`
- run `make`
make pass without any errors

I think our glide hash prevent of correct update of some packages, where can I find jenkins job code to check if adding of `make distclean` before run `make` will solve the problem?

**Comment by rmohr** [comment]
@cynepco3hahue sorry for being too unspecific. The main thing is that if the glide.lock is correct, there is neither a need to run `make distclean`, nor `make sync`.

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
I saw the same error like Jenkins. Deleting `~/.glide/cache` once solved the problem for me.

**Comment by rmohr** [comment]
@davidvossel @stu-gott, I would tend to merge this patch. Might be that some people will have to delete their glide cache because they see errors like

```
# kubevirt.io/kubevirt/vendor/k8s.io/client-go/tools/clientcmd/api
../vendor/k8s.io/client-go/tools/clientcmd/api/zz_generated.deepcopy.go:112: val.DeepCopyObject undefined (type runtime.Object has no field or method DeepCopyObject)
../vendor/k8s.io/client-go/tools/clientcmd/api/zz_generated.deepcopy.go:167: val.DeepCopyObject undefined (type runtime.Object has no field or method DeepCopyObject)
```

after we merge this.

Thoughts?

**Comment by rmohr** [comment]
The update is needed to bring in the leaderelection code for #412

**Comment by cynepco3hahue** [comment]
@davidvossel I think it some magic under the `glide`. I tried to run `glide update -v` twice, and one time it fails on some error, but the second time it succeeded without any troubles, so for me, it looks like `glide update` can not update some dependencies for the first time(maybe some import loop), but succeeds on the second time.
Also, I tried the same hack under `Makefile`, I changed `checksync` to
```
checksync:
       glide update -v
        if [ "`${HASH} glide.lock`" != "`cat .glide.lock.hash`" ]; then \
                make sync; \
        fi
```
and ran `make` twice, the first time it failed, but the second time it succeeded.

**Comment by davidvossel** [comment]
@cynepco3hahue that's really weird. Based on what we know now, it doesn't appear that there's anything we can reasonably do on our side to make this work without people having to clean glide artifacts up themselves after the change.

Here's another question. Why did we have to introduce all the DeepCopy logic?

**Comment by cynepco3hahue** [comment]
@davidvossel `runtime.Object` interface must now implement this one, you can see it under
https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/interfaces.go
Kubernetets, in this case, has some deep copy generator to add these methods automatically to all necessary object, but @rmohr said that for short-term it good enough just to implement this method by ourselves

**Comment by stu-gott** [comment]
@cynepco3hahue by the way, I've occasionally needed to run `glide cc` to clear my glide cache (in the root of $HOME, not vendor folder or source tree) in order to clear up artifacts while running `glide update -v`

File: pkg/virt-handler/virtwrap/api/schema.go [hunk 1]
```diff
@@ -143,6 +168,29 @@ type DomainList struct {
 	Items    []Domain
 }
 
+func (in *DomainList) DeepCopyInto(out *DomainList) {
+	v, _ := model.Clone(in)
+	out, _ = v.(*DomainList)
```
Comment (rmohr) [inline pkg/virt-handler/virtwrap/api/schema.go]: Could you use the cast version everywhere which will panic (`out = v.(*DomainList)`)?

File: pkg/virt-handler/virtwrap/api/schema.go [hunk 2]
```diff
@@ -143,6 +168,29 @@ type DomainList struct {
 	Items    []Domain
 }
 
+func (in *DomainList) DeepCopyInto(out *DomainList) {
+	v, _ := model.Clone(in)
```
Comment (rmohr) [inline pkg/virt-handler/virtwrap/api/schema.go]: Could you check the error and panic if an error occured?

File: pkg/virt-handler/virtwrap/api/schema.go (hunk 1 above)
Comment (cynepco3hahue) [reply pkg/virt-handler/virtwrap/api/schema.go]: :+1:

File: pkg/virt-handler/virtwrap/api/schema.go (hunk 2 above)
Comment (cynepco3hahue) [reply pkg/virt-handler/virtwrap/api/schema.go]: :+1:

**Comment by stu-gott** [comment]
I'm looking into the glide issue. I just managed to reproduce it.

**Comment by cynepco3hahue** [comment]
@stu-gott :+1:

**Comment by stu-gott** [comment]
retest this please

**Pull request merged**